signer/src/transaction_coordinator.rs

```diff
@@ -2547,7 +2547,9 @@ pub async fn should_coordinate_dkg(
     // instead of doing new DKG rounds. If we fail to do so they will eventually
     // be marked as failed, and we will resume DKG-ing.
     let latest_dkg_shares = storage.get_latest_encrypted_dkg_shares().await?;
```

Comment:
We should use a function that gets non-failed shares here. We'll have to add a query.